src/float.cr

```diff
@@ -148,13 +149,38 @@ struct Float32
     Printer.print(self, io)
   end
 
-  def hash
-    unsafe_as(Int32)
-  end
-
   def clone
     self
   end
+
+  include Number::HashNormalize
+
+  private FLOAT32_IS_IEEE754 = sizeof(Float32) == sizeof(UInt32) &&
+                               1.0_f32.unsafe_as(UInt32) == 0x3f800000_u32 &&
+                               -1.5009765625_f32.unsafe_as(UInt32) == 0xbfc02000_u32
+
+  def hash_normalize
+    float_normalize_wrap do
+      {% if FLOAT32_IS_IEEE754 %}
```

Comment:
I could revert it back to `flag?(:i686) || flag?(:x86_64) || flag?(:arm) || flag?(:aarch64)` - that will be reliable for cross-compilation.

I don't want to leave it for runtime check cause of performance.
If performance doesn't matter, then this bitwise trick isn't needed at all.
But I strongly think, performance matters.